strelets/adapter/swarm_volumes.go

```diff
@@ -64,5 +55,43 @@ func (d *dockerSwarm) provisionVolume(ctx context.Context, volumeName string, ta
 		Type:     "volume",
 		Target:   target,
 		ReadOnly: false,
+		VolumeOptions: &mount.VolumeOptions{
+			DriverConfig: &mount.Driver{
+				Name:    driverName,
+				Options: driverOptions,
+			},
+		},
 	}, nil
 }
+
+func getVolumeDriverOptions(volumeName string, orchestratorOptions OrchestratorOptions, maxSizeInGb int) (string, map[string]string) {
+	driver := LOCAL_DRIVER
+
+	if orchestratorOptions.StorageDriver != "" {
+		driver = orchestratorOptions.StorageDriver
+	}
+
+	driverOptions := make(map[string]string)
+	for k, v := range orchestratorOptions.StorageOptions {
+		driverOptions[k] = v
+	}
+
+	// Only enable size option for supported drivers
+	switch driver {
+	case REXRAY_EBS_DRIVER:
+		driverOptions["size"] = strconv.Itoa(maxSizeInGb)
+	case LOCAL_DRIVER:
+		if fsType, ok := driverOptions["type"]; ok && fsType == "nfs" {
+			// append volumeName to the common shared volume storage directory
+			dir := driverOptions["device"] + "/" + volumeName
+			driverOptions["device"] = dir
+			// Warning: we assume that the volume directory exists on this machine, or its parent is mounted
+			if strings.HasPrefix(dir, "/") {
```

Comment:
Should be `strings.HasPrefix(dir, ":")`, because device usually starts with ":", as seen for example here, at the bottom of the page: https://docs.docker.com/engine/reference/commandline/volume_create/